Build fails with qt 6.10.1

**Describe the bug**
In ArchLinux `qt` packages (`qt6-base` `qt6-svg` `qt6-imageformats` `qt6-tools`) have been updated to version 6.10.1. And now build fails:
<details>
  <summary>Spoiler</summary>
  
  ```
==> Starting build()...
-- The CXX compiler identification is GNU 15.2.1
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /usr/bin/c++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD - Success
-- Found Threads: TRUE
-- Performing Test HAVE_STDATOMIC
-- Performing Test HAVE_STDATOMIC - Success
-- Found WrapAtomic: TRUE
-- Found OpenGL: /usr/lib/libOpenGL.so
-- Found WrapOpenGL: TRUE
-- Could NOT find WrapVulkanHeaders (missing: Vulkan_INCLUDE_DIR)
-- Found ICU: /usr/include (found version "78.1") found components: uc i18n
-- Found Taglib: /usr/lib/libtag.so (found version "2.1.1")
-- Found PkgConfig: /usr/bin/pkg-config (found version "2.5.1")
-- Checking for module 'libavcodec'
--   Found libavcodec, version 62.11.100
-- Checking for module 'libavformat'
--   Found libavformat, version 62.3.100
-- Checking for module 'libavdevice'
--   Found libavdevice, version 62.1.100
-- Checking for module 'libavutil'
--   Found libavutil, version 60.8.100
-- Checking for module 'libavfilter'
--   Found libavfilter, version 11.4.100
-- Checking for module 'libswscale'
--   Found libswscale, version 9.1.100
-- Checking for module 'libswresample'
--   Found libswresample, version 6.1.100
-- Checking for module 'libpostproc'
--   Package 'libpostproc' not found
-- Found ZLIB: /usr/lib/libz.so (found version "1.3.1")
-- Found FFmpeg: /usr/lib/libavcodec.so;/usr/lib/libavfilter.so;/usr/lib/libavformat.so;/usr/lib/libavutil.so;/usr/lib/libswresample.so
-- Using 3rd-party KDSingleApplication
-- Using 3rd-party QCoro
-- LIBGME not found; skipping gme plugin.
--
-- ******** Fooyin Configuration ********
-- General:
--   Fooyin version        : 0.9.2
--   CMake version         : 4.2.0
--   CMake command         : /usr/bin/cmake
--   System                : Linux
--   C++ compiler          : /usr/bin/c++
--   C++ compiler version  : 15.2.1
--   CXX flags             : -O3 -march=native -flto -fuse-linker-plugin -fcoroutines
--   Build type            : Release
--
-- Install location:
--   Install prefix        : /usr/local
--   Binary install dir    : bin
--   Library install dir   : lib/fooyin
--   Plugin install dir    : lib/fooyin/plugins
--
-- Options:
--   BUILD_SHARED_LIBS     : ON
--   BUILD_TESTING         : OFF
--   BUILD_PLUGINS         : ON
--   BUILD_ALSA            : ON
--   BUILD_TRANSLATIONS    : ON
--   BUILD_CCACHE          : ON
--   BUILD_PCH             : OFF
--   BUILD_WERROR          : OFF
--   BUILD_ASAN            : OFF
--   INSTALL_FHS           : ON
--   INSTALL_HEADERS       : OFF
-- Configuring done (8.4s)
-- Generating done (0.4s)
-- Build files have been written to: /home/alexey/git/arch-i3-config/custom_builds/fooyin/src/fooyin-custom-build/build
[5/521] Generating data/fooyin_af.qm
Updating '/home/alexey/git/arch-i3-config/custom_builds/fooyin/src/fooyin-custom-build/build/data/fooyin_af.qm'...
    Generated 110 translation(s) (29 finished and 81 unfinished)
    Ignored 1486 untranslated source text(s)
[6/521] Generating data/fooyin_ca.qm
Updating '/home/alexey/git/arch-i3-config/custom_builds/fooyin/src/fooyin-custom-build/build/data/fooyin_ca.qm'...
    Generated 1542 translation(s) (1480 finished and 62 unfinished)
    Ignored 54 untranslated source text(s)
[7/521] Generating data/fooyin_de.qm
Updating '/home/alexey/git/arch-i3-config/custom_builds/fooyin/src/fooyin-custom-build/build/data/fooyin_de.qm'...
    Generated 1582 translation(s) (1574 finished and 8 unfinished)
    Ignored 14 untranslated source text(s)
[8/521] Generating data/fooyin_en.qm
Updating '/home/alexey/git/arch-i3-config/custom_builds/fooyin/src/fooyin-custom-build/build/data/fooyin_en.qm'...
    Generated 0 translation(s) (0 finished and 0 unfinished)
    Ignored 1596 untranslated source text(s)
[9/521] Generating data/fooyin_en_GB.qm
Updating '/home/alexey/git/arch-i3-config/custom_builds/fooyin/src/fooyin-custom-build/build/data/fooyin_en_GB.qm'...
    Generated 1 translation(s) (1 finished and 0 unfinished)
[10/521] Generating data/fooyin_en_US.qm
Updating '/home/alexey/git/arch-i3-config/custom_builds/fooyin/src/fooyin-custom-build/build/data/fooyin_en_US.qm'...
    Generated 1540 translation(s) (1515 finished and 25 unfinished)
    Ignored 56 untranslated source text(s)
[11/521] Generating data/fooyin_es.qm
Updating '/home/alexey/git/arch-i3-config/custom_builds/fooyin/src/fooyin-custom-build/build/data/fooyin_es.qm'...
    Generated 747 translation(s) (453 finished and 294 unfinished)
    Ignored 849 untranslated source text(s)
[12/521] Generating data/fooyin_fi.qm
Updating '/home/alexey/git/arch-i3-config/custom_builds/fooyin/src/fooyin-custom-build/build/data/fooyin_fi.qm'...
    Generated 244 translation(s) (86 finished and 158 unfinished)
    Ignored 1352 untranslated source text(s)
[13/521] Generating data/fooyin_fr.qm
Updating '/home/alexey/git/arch-i3-config/custom_builds/fooyin/src/fooyin-custom-build/build/data/fooyin_fr.qm'...
    Generated 1541 translation(s) (1505 finished and 36 unfinished)
    Ignored 55 untranslated source text(s)
[14/521] Generating data/fooyin_id.qm
Updating '/home/alexey/git/arch-i3-config/custom_builds/fooyin/src/fooyin-custom-build/build/data/fooyin_id.qm'...
    Generated 204 translation(s) (64 finished and 140 unfinished)
    Ignored 1392 untranslated source text(s)
[15/521] Generating data/fooyin_ie.qm
Updating '/home/alexey/git/arch-i3-config/custom_builds/fooyin/src/fooyin-custom-build/build/data/fooyin_ie.qm'...
    Generated 455 translation(s) (285 finished and 170 unfinished)
    Ignored 1141 untranslated source text(s)
[16/521] Generating data/fooyin_it.qm
Updating '/home/alexey/git/arch-i3-config/custom_builds/fooyin/src/fooyin-custom-build/build/data/fooyin_it.qm'...
    Generated 200 translation(s) (69 finished and 131 unfinished)
    Ignored 1396 untranslated source text(s)
[17/521] Generating data/fooyin_ja.qm
Updating '/home/alexey/git/arch-i3-config/custom_builds/fooyin/src/fooyin-custom-build/build/data/fooyin_ja.qm'...
    Generated 544 translation(s) (287 finished and 257 unfinished)
    Ignored 1052 untranslated source text(s)
[18/521] Generating data/fooyin_lt.qm
Updating '/home/alexey/git/arch-i3-config/custom_builds/fooyin/src/fooyin-custom-build/build/data/fooyin_lt.qm'...
    Generated 680 translation(s) (404 finished and 276 unfinished)
    Ignored 916 untranslated source text(s)
[19/521] Generating data/fooyin_nb_NO.qm
Updating '/home/alexey/git/arch-i3-config/custom_builds/fooyin/src/fooyin-custom-build/build/data/fooyin_nb_NO.qm'...
    Generated 343 translation(s) (140 finished and 203 unfinished)
    Ignored 1253 untranslated source text(s)
[20/521] Generating data/fooyin_nl.qm
Updating '/home/alexey/git/arch-i3-config/custom_builds/fooyin/src/fooyin-custom-build/build/data/fooyin_nl.qm'...
    Generated 779 translation(s) (519 finished and 260 unfinished)
    Ignored 817 untranslated source text(s)
[21/521] Generating data/fooyin_pl.qm
Updating '/home/alexey/git/arch-i3-config/custom_builds/fooyin/src/fooyin-custom-build/build/data/fooyin_pl.qm'...
    Generated 808 translation(s) (429 finished and 379 unfinished)
    Ignored 788 untranslated source text(s)
[22/521] Generating data/fooyin_pt.qm
Updating '/home/alexey/git/arch-i3-config/custom_builds/fooyin/src/fooyin-custom-build/build/data/fooyin_pt.qm'...
    Generated 798 translation(s) (595 finished and 203 unfinished)
    Ignored 798 untranslated source text(s)
[24/521] Generating data/fooyin_pt_BR.qm
Updating '/home/alexey/git/arch-i3-config/custom_builds/fooyin/src/fooyin-custom-build/build/data/fooyin_pt_BR.qm'...
    Generated 1127 translation(s) (1069 finished and 58 unfinished)
    Ignored 469 untranslated source text(s)
[36/521] Building CXX object src/utils/CMakeFiles/fooyin_utils.dir/fypaths.cpp.o
/home/alexey/git/arch-i3-config/custom_builds/fooyin/src/fooyin-custom-build/src/utils/fypaths.cpp:40:9: warning: ‘QString {anonymous}::baseDir()’ defined but not used [-Wunused-function]
   40 | QString baseDir()
      |         ^~~~~~~
[38/521] Building CXX object src/utils/CMakeFiles/fooyin_utils.dir/starrating.cpp.o
FAILED: [code=1] src/utils/CMakeFiles/fooyin_utils.dir/starrating.cpp.o
/usr/bin/c++ -DQT_CONCURRENT_LIB -DQT_CORE_LIB -DQT_DISABLE_DEPRECATED_BEFORE=0x060200 -DQT_GUI_LIB -DQT_NO_CAST_FROM_ASCII -DQT_NO_CAST_FROM_BYTEARRAY -DQT_NO_CAST_TO_ASCII -DQT_NO_DEBUG -DQT_NO_NARROWING_CONVERSIONS_IN_CONNECT -DQT_SQL_LIB -DQT_STRICT_ITERATORS -DQT_USE_QSTRINGBUILDER -DQT_WIDGETS_LIB -Dfooyin_utils_EXPORTS -I/home/alexey/git/arch-i3-config/custom_builds/fooyin/src/fooyin-custom-build/build/src/utils/fooyin_utils_autogen/include -I/home/alexey/git/arch-i3-config/custom_builds/fooyin/src/fooyin-custom-build/src/utils -I/home/alexey/git/arch-i3-config/custom_builds/fooyin/src/fooyin-custom-build/include -isystem /home/alexey/git/arch-i3-config/custom_builds/fooyin/src/fooyin-custom-build/build/src/utils -isystem /usr/include/qt6/QtCore -isystem /usr/include/qt6 -isystem /usr/lib/qt6/mkspecs/linux-g++ -isystem /usr/include/qt6/QtWidgets -isystem /usr/include/qt6/QtGui -isystem /usr/include/qt6/QtSql -isystem /usr/include/qt6/QtConcurrent -O3 -march=native -flto -fuse-linker-plugin -fcoroutines -O3 -DNDEBUG -std=gnu++23 -fPIC -fvisibility=hidden -fvisibility-inlines-hidden -Wall -Wextra -Wpedantic -Wunused -Wshadow -Wundef -Wuninitialized -Wredundant-decls -Wcast-align -Winit-self -Wmissing-include-dirs -Wstrict-overflow=2 -Wunused-parameter -Wdisabled-optimization -Woverloaded-virtual -Wold-style-cast -Wno-array-bounds -mno-direct-extern-access -MD -MT src/utils/CMakeFiles/fooyin_utils.dir/starrating.cpp.o -MF src/utils/CMakeFiles/fooyin_utils.dir/starrating.cpp.o.d -o src/utils/CMakeFiles/fooyin_utils.dir/starrating.cpp.o -c /home/alexey/git/arch-i3-config/custom_builds/fooyin/src/fooyin-custom-build/src/utils/starrating.cpp
/home/alexey/git/arch-i3-config/custom_builds/fooyin/src/fooyin-custom-build/src/utils/starrating.cpp: In member function ‘void Fooyin::StarRating::paint(QPainter*, const QRect&, const QPalette&, EditMode, Qt::Alignment) const’:
/home/alexey/git/arch-i3-config/custom_builds/fooyin/src/fooyin-custom-build/src/utils/starrating.cpp:107:38: error: no matching function for call to ‘QString::arg(Qt::Alignment&)’
  102 |     const QString cacheKey = u"StarRating:%1|%2|%3|%4|%5|%6"_s.arg(m_rating)
      |                              ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  103 |                                  .arg(m_scale)
      |                                  ~~~~~~~~~~~~~
  104 |                                  .arg(m_maxCount)
      |                                  ~~~~~~~~~~~~~~~~
  105 |                                  .arg(mode == EditMode::Editable ? 1 : 0)
      |                                  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  106 |                                  .arg(rect.width())
      |                                  ~~~~~~~~~~~~~~~~~~
  107 |                                  .arg(alignment);
      |                                  ~~~~^~~~~~~~~~~
/home/alexey/git/arch-i3-config/custom_builds/fooyin/src/fooyin-custom-build/src/utils/starrating.cpp:107:38: note: there are 4 candidates
In file included from /usr/include/qt6/QtCore/qhashfunctions.h:9,
                 from /usr/include/qt6/QtCore/qfloat16.h:10,
                 from /usr/include/qt6/QtCore/qmetatype.h:15,
                 from /usr/include/qt6/QtCore/QMetaType:1,
                 from /home/alexey/git/arch-i3-config/custom_builds/fooyin/src/fooyin-custom-build/include/utils/starrating.h:24,
                 from /home/alexey/git/arch-i3-config/custom_builds/fooyin/src/fooyin-custom-build/src/utils/starrating.cpp:20:
/usr/include/qt6/QtCore/qstring.h:324:27: note: candidate 1: ‘template<class T, typename std::enable_if<conjunction_v<std::disjunction<std::is_integral<_Tp>, std::conjunction<std::is_enum<_Tp>, std::negation<std::is_scoped_enum<T> > > >, std::negation<std::disjunction<std::is_same<typename std::remove_cvref<_Tp>::type, _Float16>, std::is_same<typename std::remove_cvref<_Tp>::type, qfloat16>, std::is_floating_point<_Tp> > >, std::negation<std::conjunction<std::negation<QtPrivate::treat_as_integral_arg<typename std::remove_cv< <template-parameter-1-1> >::type> >, std::is_convertible<T, QAnyStringView> > > >, bool>::type <anonymous> > QString QString::arg(T, int, int, QChar) const’
  324 |     [[nodiscard]] QString arg(T a, int fieldWidth = 0, int base = 10,
      |                           ^~~
/usr/include/qt6/QtCore/qstring.h:324:27: note: template argument deduction/substitution failed:
/usr/include/qt6/QtCore/qstring.h:323:53: error: no type named ‘type’ in ‘struct std::enable_if<false, bool>’
  323 |     template <typename T, if_integral_non_char<T> = true>
      |                                                     ^~~~
/usr/include/qt6/QtCore/qstring.h:339:27: note: candidate 2: ‘template<class T, typename std::enable_if<std::disjunction<std::is_same<typename std::remove_cvref<_Tp>::type, _Float16>, std::is_same<typename std::remove_cvref<_Tp>::type, qfloat16>, std::is_floating_point<_Tp> >::value, bool>::type <anonymous> > QString QString::arg(T, int, char, int, QChar) const’
  339 |     [[nodiscard]] QString arg(T a, int fieldWidth = 0, char format = 'g', int precision = -1,
      |                           ^~~
/usr/include/qt6/QtCore/qstring.h:339:27: note: template argument deduction/substitution failed:
/usr/include/qt6/QtCore/qstring.h:338:50: error: no type named ‘type’ in ‘struct std::enable_if<false, bool>’
  338 |     template <typename T, if_floating_point<T> = true>
      |                                                  ^~~~
/usr/include/qt6/QtCore/qstring.h:344:27: note: candidate 3: ‘template<class T, typename std::enable_if<std::conjunction<std::negation<QtPrivate::treat_as_integral_arg<typename std::remove_cv< <template-parameter-1-1> >::type> >, std::is_convertible<T, QAnyStringView> >::value, bool>::type <anonymous> > QString QString::arg(const T&, int, QChar) const’
  344 |     [[nodiscard]] QString arg(const T &a, int fieldWidth = 0, QChar fillChar = u' ') const
      |                           ^~~
/usr/include/qt6/QtCore/qstring.h:344:27: note: template argument deduction/substitution failed:
/usr/include/qt6/QtCore/qstring.h:343:47: error: no type named ‘type’ in ‘struct std::enable_if<false, bool>’
  343 |     template <typename T, if_string_like<T> = true>
      |                                               ^~~~
/usr/include/qt6/QtCore/qstring.h:364:5: note: candidate 4: ‘template<class ... Args> typename std::enable_if<((sizeof... (Args) >= 2) && conjunction_v<std::conjunction<std::negation<QtPrivate::treat_as_integral_arg<typename std::remove_cv<Args>::type> >, std::is_convertible<Args, QAnyStringView> >...>), QString>::type QString::arg(Args&& ...) const’
  364 |     arg(Args &&...args) const
      |     ^~~
/usr/include/qt6/QtCore/qstring.h:364:5: note: template argument deduction/substitution failed:
/usr/include/qt6/QtCore/qstring.h: In substitution of ‘template<class ... Args> typename std::enable_if<((sizeof... (Args) >= 2) && conjunction_v<std::conjunction<std::negation<QtPrivate::treat_as_integral_arg<typename std::remove_cv<Args>::type> >, std::is_convertible<Args, QAnyStringView> >...>), QString>::type QString::arg(Args&& ...) const [with Args = {QFlags<Qt::AlignmentFlag>&}]’:
/home/alexey/git/arch-i3-config/custom_builds/fooyin/src/fooyin-custom-build/src/utils/starrating.cpp:107:38:   required from here
  102 |     const QString cacheKey = u"StarRating:%1|%2|%3|%4|%5|%6"_s.arg(m_rating)
      |                              ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  103 |                                  .arg(m_scale)
      |                                  ~~~~~~~~~~~~~
  104 |                                  .arg(m_maxCount)
      |                                  ~~~~~~~~~~~~~~~~
  105 |                                  .arg(mode == EditMode::Editable ? 1 : 0)
      |                                  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  106 |                                  .arg(rect.width())
      |                                  ~~~~~~~~~~~~~~~~~~
  107 |                                  .arg(alignment);
      |                                  ~~~~^~~~~~~~~~~
/usr/include/qt6/QtCore/qstring.h:364:5: error: no type named ‘type’ in ‘struct std::enable_if<false, QString>’
  364 |     arg(Args &&...args) const
      |     ^~~
[40/521] Building CXX object src/utils/CMakeFiles/fooyin_utils.dir/stardelegate.cpp.o
ninja: build stopped: subcommand failed.
==> ERROR: A failure occurred in build().
    Aborting...
  ```
  
</details>

**To Reproduce**
```
cmake \
  -DCMAKE_CXX_STANDARD=23 \
  -DCMAKE_CXX_FLAGS="-O3 -march=native -flto -fuse-linker-plugin" \
  -S . \
  -G Ninja \
  -B build &&
  cmake --build ./build -j 3
```

**System:**
 - OS: ArchLinux
 - fooyin version: master branch

**Additional context**
After downgrading `qt` packages to 6.10.0 build succeeds with no errors.

--- a/src/utils/starrating.cpp
+++ b/src/utils/starrating.cpp
@@ -104,7 +104,7 @@ void StarRating::paint(QPainter* painter, const QRect& rect, const QPalette& pal
                                  .arg(m_maxCount)
                                  .arg(mode == EditMode::Editable ? 1 : 0)
                                  .arg(rect.width())
-                                 .arg(alignment);
+                                 .arg(alignment.toInt());
 
     QPixmap pixmap;
     if(!QPixmapCache::find(cacheKey, &pixmap)) {

